common-rest/src/main/java/org/uniprot/api/rest/service/BasicSearchService.java

```diff
@@ -59,37 +89,84 @@ public T getEntity(String idField, String value) {
         }
     }
 
-    public QueryResult<T> search(SolrRequest request, String cursor, int pageSize) {
-        QueryResult<R> results = repository.searchPage(request, cursor, pageSize);
-        List<T> converted =
+    public QueryResult<R> search(SearchRequest request) {
+        SolrRequest solrRequest = createSearchSolrRequest(request);
+
+        QueryResult<D> results = repository.searchPage(solrRequest, request.getCursor());
+        List<R> converted =
                 results.getContent().stream()
                         .map(entryConverter)
                         .filter(Objects::nonNull)
                         .collect(Collectors.toList());
         return QueryResult.of(converted, results.getPage(), results.getFacets());
     }
 
-    public Stream<T> download(SolrRequest request) {
-        return repository.getAll(request).map(entryConverter).filter(Objects::nonNull);
+    public Stream<R> download(SearchRequest request) {
+        SolrRequest solrRequest = createDownloadSolrRequest(request);
+
+        return repository
+                .getAll(solrRequest)
+                .map(entryConverter)
+                .filter(Objects::nonNull)
+                .limit(solrRequest.getTotalRows());
     }
 
-    public SolrRequest createSolrRequest(
-            SearchRequest request,
-            FacetConfig facetConfig,
-            AbstractSolrSortClause solrSortClause,
-            DefaultSearchHandler defaultSearchHandler) {
+    /*
+    to create request for search api.
+    include facets true for search api
+    */
+    public SolrRequest createSearchSolrRequest(SearchRequest request) {
+        return createSearchSolrRequest(request, true);
+    }
+
+    /*
+     // case 1. size is not passed, use  DEFAULT_RESULTS_SIZE(25) then set rows and totalCount as DEFAULT_RESULTS_SIZE
+    // case 2. size is less than or equal solrBatchSize(10,000 or 100) then set SolrRequest's rows, totalCount value to size
+    // case 3. size is greater than solrBatchSize(10,000 or 100) then then set SolrRequest's rows, totalCount value to solrBatchSize
+    // for the search rows, size and total rows should be same. We should restrict size value less than solrBatchSize
```

Comment:
could you align the comments please?

Reply:
Also, in the comments it refers to `totalCount` in SolrRequest. It should be `totalRows`